### ISSUE TITLE
NullReferenceException terminates dotnet process

### ISSUE THREAD

**Issue by martincostello** [NLog/NLog#1703]
**Type**: Bug

**NLog version**: 4.2.4

**Platform**: .NET Core 

**Current NLog config**:

``` xml
<?xml version="1.0" encoding="utf-8" ?>
<nlog xmlns="http://www.nlog-project.org/schemas/NLog.xsd"
      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
      throwExceptions="true"
      autoReload="true">
  <extensions>
    <add assembly="NLog.Web.AspNetCore" />
    <add assembly="MyFeature" />
  </extensions>
  <variable name="feature_name" value="MyFeature" />
  <variable name="sortabledate" value="${date:format=yyyyMMddHH00}" />
  <variable name="log_dir" value="${basedir}/../logs" />
  <variable name="archive_dir" value="${log_dir}/logs_archive" />
  <variable name="human_readable_layout" value="${longdate}|${level}|${logger}|${message}|${onexception:EXCEPTION OCCURRED\:${exception:format=type,message,method,stackTrace:maxInnerExceptionLevel=5:innerFormat=shortType,message,method,stackTrace}}" />
  <targets async="true">
    <target xsi:type="Null" name="drop" />
    <target name="nxlog_json"
            fileName="${log_dir}\nxlog\${level:lowercase=true}.log"
            xsi:type="File"
            encoding="utf-8"
            archiveEvery="Hour"
            archiveNumbering="Date"
            archiveDateFormat="yyyyMMdd-HHmm"
            archiveFileName="${archive_dir}\nxlog\{#####}-${level:lowercase=true}.log"
            maxArchiveFiles="12"
            concurrentWrites="false"
            keepFileOpen="false">
      <layout xsi:Type="flattenedjsonlayout">
        <attribute name="Request" layout="${aspnet-request:header=x-request}" />
        <attribute name="Conversation" layout="${aspnet-request:header=x-conversation}" />
        <attribute name="Session" layout="${aspnet-request:header=x-session}" />
        <attribute name="User" layout="${aspnet-request:header=x-user}" />
        <attribute name="TrafficType" layout="${aspnet-request:header=x-traffictype}" />
        <attribute name="UserAgent" layout="${aspnet-request:header=user-agent}" />
      </layout>
    </target>
    <target name="human_text_file"
            fileName="${log_dir}\human_readable\human_readable.log"
            xsi:type="File"
            encoding="utf-8"
            layout="${human_readable_layout}"
            archiveEvery="Hour"
            archiveNumbering="Date"
            archiveDateFormat="yyyyMMdd-HHmm"
            archiveFileName="${archive_dir}\human_readable\{#####}-human_readable.log"
            maxArchiveFiles="36"
            concurrentWrites="false"
            keepFileOpen="false" />
  </targets>
  <rules>
    <logger name="*" minlevel="Debug" writeTo="nxlog_json" />
    <logger name="*" minlevel="Debug" writeTo="human_text_file" />
  </rules>
</nlog>

```
- What is the current result? Logging does not crash the dotnet process.
- What is the expected result? Logging crashes the dotnet process.

```
Unhandled Exception: System.NullReferenceException: Object reference not set to an instance of an object.
   at NLog.LayoutRenderers.ExceptionLayoutRenderer.ParseMethodNameFromStackTrace(String stackTrace)
   at NLog.LayoutRenderers.ExceptionLayoutRenderer.AppendMethod(StringBuilder sb, Exception ex)
   at NLog.LayoutRenderers.ExceptionLayoutRenderer.Append(StringBuilder builder, LogEventInfo logEvent)
   at NLog.LayoutRenderers.LayoutRenderer.Render(StringBuilder builder, LogEventInfo logEvent)
   at NLog.Layouts.SimpleLayout.GetFormattedMessage(LogEventInfo logEvent)
   at NLog.LayoutRenderers.Wrappers.OnExceptionLayoutRendererWrapper.RenderInner(LogEventInfo logEvent)
   at NLog.LayoutRenderers.Wrappers.WrapperLayoutRendererBase.Append(StringBuilder builder, LogEventInfo logEvent)
   at NLog.LayoutRenderers.LayoutRenderer.Render(StringBuilder builder, LogEventInfo logEvent)
   at NLog.Layouts.SimpleLayout.GetFormattedMessage(LogEventInfo logEvent)
   at NLog.Targets.FileTarget.GetFormattedMessage(LogEventInfo logEvent)
   at NLog.Targets.FileTarget.GetBytesToWrite(LogEventInfo logEvent)
   at NLog.Targets.FileTarget.Write(AsyncLogEventInfo[] logEvents)
   at NLog.Targets.Target.WriteAsyncLogEvents(AsyncLogEventInfo[] logEvents)
   at NLog.Targets.Wrappers.AsyncTargetWrapper.ProcessPendingEvents(Object state)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.TimerQueueTimer.CallCallback()
   at System.Threading.TimerQueueTimer.Fire()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
```

I haven't been reliably able to create the crash, but it seems to happen under high load. From looking at the source code at a guess there's an exception that for some reason doesn't have a stack trace, causing a `NullReferenceException` when trying to [parse the method name](https://github.com/NLog/NLog/blob/master/src/NLog/LayoutRenderers/ExceptionLayoutRenderer.cs#L365) from it.

**Comment by 304NotModified** [comment]
This is a bug, not sure if this one is already fixed. 

Any way, never set throwExceptions="true" in production!

**Comment by daniefer** [comment]
I get this exception too, but it does not crash my app. However, I did narrow it down to the code `${stacktrace:format=DetailedFlat}` and `${exception:innerFormat=Message,Type,ToString,StackTrace:maxInnerExceptionLevel=3:innerExceptionSeparator=\r\nInnerExceptions:\r\n:separator=\r\n:format=ToString}`  in my layout. As soon as I took it out, the problem went away.

**Comment by 304NotModified** [comment]
Which version?

**Comment by daniefer** [comment]
@304NotModified Updated comment with version info

**Comment by 304NotModified** [comment]
Thanks!

**Comment by 304NotModified** [comment]
@daniefer is this also .NET Core?

**Comment by daniefer** [comment]
@304NotModified - No, this is for .NET 4.5. I don't have time now to get the stacktrace for you, but I will later today or this week.

**Comment by 304NotModified** [comment]
Ok, that would be great!

**Comment by 304NotModified** [comment]
@martincostello 4.2.4 don't runs on .NET Core. Do you mean ASP.NET Core on NET46?

**Comment by martincostello** [comment]
@304NotModified The stack trace lead me here, but these are the packages I'm consuming:

```json
"NLog.Extensions.Logging": "1.0.0-rtm-alpha4",
"NLog.Web.AspNetCore": "4.2.4",
```

**Comment by 304NotModified** [comment]
ah that's "NLog.Web.AspNetCore": "4.2.4" and not "NLog.Web: "4.2.4"

**Comment by 304NotModified** [comment]
Could you check the NLog version?

**Comment by 304NotModified** [comment]
The NPE has been fixed in NLog 4.3.11. This will go into NLog 5 in the next beta release.
